lib/lens/macros.ex

```diff
@@ -13,7 +13,16 @@ defmodule Lens.Macros do
     end
 
     quote do
-      def unquote(header), do: unquote(body)
+      def unquote(header) do
+        lens = unquote(body)
+        fn
+          :get, data, next ->
```

Comment:
This is pretty neat. However, it has the side effect of making it worse suited for a public API. I envisioned clients using `deflens` in their own projects to get `|>` for free, but now that's not as simple (because of `wrap`). It seems there should be 2 macros - one to build this `:get/:get_and_update` fork internally and another one to expose to the world and use in cases like

```elixir
  deflens all, do: wrap filter(fn _ -> true end)
```

so that `wrap` is not needed.